module3/homework/vectorOfSharedPointers/vectorFunctions.cpp

```
@@ -0,0 +1,55 @@
+#include "vectorFunctions.hpp"
+#include <iostream>
+#include <algorithm>
+
+std::vector<std::shared_ptr<int>> generate(int count)
+{
+	std::vector<std::shared_ptr<int>> vectorToReturn;
+	vectorToReturn.reserve(count);
+
+	for (size_t i = 0; i < count; i++) 
+	{
+		vectorToReturn.emplace_back(std::make_shared<int>(i));
+	}
+
+	return vectorToReturn;
+}
+
+void print(const std::vector<std::shared_ptr<int>>& vectorToPrint)
+{
+	for(auto const& it : vectorToPrint)
+	{
+		std::cout << *it;
+	}
+}
+
+void add10(const std::vector<std::shared_ptr<int>>& vec)
+{
+	for(auto& it : vec)
+	{
+		if(!it)
+		{
+			continue;
+		}
+
+		*it += 10;
```

```suggestion
		if(it)
		{
			*it += 10;
		}
```

---

why do you prefer this version? I used to stick to the principle fail faster, this is why you usually find the actual thing that happens at the bottom of the method